File: Week 3/Python Codes/ReverseLinkedList.py

# Node Class
class node:
    def __init__(self, val):
        self.data = val
        self.next = None

# This function should reverse linked list and returns head of the modified linked list
def reverseList(head):
    prev, curr, next = None, head, None
    while curr is not None:
        next = curr.next
        curr.next = prev
        prev = curr
        curr = next
    head = prev
    return head

File: Week 3/C++ Codes/ReverseLinkedList.cpp
#include<bits/stdc++.h>
using namespace std;

// Linked List Node structure:
struct Node
{
    int data;
    struct Node *next;
};

// Should reverse list and return new head
struct Node* reverseList(struct Node *head)
{
    struct Node* prev = NULL;
    struct Node* curr = head;
    struct Node* next = NULL;
    while (curr != NULL) {
        next = curr->next;
        curr->next = prev;
        prev = curr;
        curr = next;
    }
    head = prev;
    return head;
}